portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,8 +62,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String username = request.getParameter("username");
     String message = request.getParameter("comment-or-question");
-    messages.put(username, message);
-    // messages.add(username);
+    String maxNumCommentsString = request.getParameter("max-num-comments");
+    try {
+      maxNumComments = Integer.parseInt(maxNumCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxNumCommentsString);
+      System.err.println("Using default value of " + defaultMaxNumComments + ".");
```

Comment:
An alternative approach to format strings is to use the `String.format` java method. [Here are some examples](https://dzone.com/articles/java-string-format-examples). It'll give you more control and in some cases it's less error prone.

Reply:
Ok, I'll adjust the code to use String.format.